### PR TITLE
Allow tls=false override with mongodb+srv protocol

### PR DESCRIPTION
**Pull request by jyemin**
JAVA-4915

### CODE REVIEW

File: driver-core/src/test/unit/com/mongodb/ConnectionStringSpecification.groovy [hunk 1]
```diff
@@ -683,6 +683,20 @@ class ConnectionStringSpecification extends Specification {
...
+    def 'should set sslEnabled property with SRV protocol'() {
+        expect:
+        connectionString.getSslEnabled() == sslEnabled
+
+        where:
+        connectionString                                                           | sslEnabled
+        new ConnectionString('mongodb+srv://test5.test.build.10gen.cc')            | true
+        new ConnectionString('mongodb+srv://test5.test.build.10gen.cc/?tls=true')  | true
+        new ConnectionString('mongodb+srv://test5.test.build.10gen.cc/?ssl=true')  | true
+        new ConnectionString('mongodb+srv://test5.test.build.10gen.cc/?tls=false') | false
```
Comment (jyemin) [inline driver-core/src/test/unit/com/mongodb/ConnectionStringSpecification.groovy:696]: This is the only one that failed.  The others are just to increase test coverage.

File: driver-core/src/main/com/mongodb/ConnectionString.java [hunk 2]
```diff
@@ -420,8 +420,8 @@ public ConnectionString(final String connectionString, @Nullable final DnsClient
             + "'%s' contains the keys %s", ALLOWED_OPTIONS_IN_TXT_RECORD, unresolvedHosts.get(0), txtRecordsOptionsMap.keySet()));
         }
         Map<String, List<String>> combinedOptionsMaps = combineOptionsMaps(txtRecordsOptionsMap, connectionStringOptionsMap);
-        if (isSrvProtocol && !combinedOptionsMaps.containsKey("ssl")) {
-            combinedOptionsMaps.put("ssl", singletonList("true"));
+        if (isSrvProtocol && !(combinedOptionsMaps.containsKey("tls") || combinedOptionsMaps.containsKey("ssl"))) {
+            combinedOptionsMaps.put("tls", singletonList("true"));
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/ConnectionString.java:424]: Switched to the non-deprecated parameter name

Comment (rozza) [reply driver-core/src/main/com/mongodb/ConnectionString.java:424]: Ack

**Review by rozza — APPROVED** [review]
LGTM - nice and subtle bug that one.

**Pull request merged**